elastic/kibana
# [Dashboard Navigation] Simplify state management

Closes https://github.com/elastic/kibana/issues/167577

## Summary

Previously, the Link embeddable used the whole redux embeddable package - however, the overall state that needs to be managed for this panel is very simple, so this ended up being overkill. This PR fixes that by adding a `useLinksAttributes` hook to replace the redux package that subscribes to changes made to the attributes instead. 

I also made two smaller changes in this PR:
1. Called the "Organize imports" command from VSCode on all of the touched files - this explains all of the seemingly unrelated import changes.
2. I fixed the React warning that was being thrown due to calling `setIsSaving` after the component was unmounted.

### How to Test

To test number 2 above, create a by-reference Links panel and refresh the dashboard. Then,
1. Make some sort of change to the Links panel, such as re-arranging the links
2. Save the changes - note that, without the mount check, the following React error will be thrown:
   ![image](https://github.com/elastic/kibana/assets/8698078/88573c7b-8469-490d-83dd-5e335573aa75)
3. Now, with the mount check, this no longer happens 🎉 

### Checklist


### For maintainers
